spring-integration-ip/src/main/java/org/springframework/integration/ip/udp/UnicastSendingMessageHandler.java

```diff
@@ -381,7 +407,11 @@ public int getSoReceiveBufferSize() {
 	@Override
 	protected void onInit() throws Exception {
 		super.onInit();
-		this.mapper.setBeanFactory(this.getBeanFactory());
+		this.mapper.setBeanFactory(getBeanFactory());
+		this.evaluationContext = IntegrationContextUtils.getEvaluationContext(getBeanFactory());
+		if (this.socketExpression != null) {
+			Assert.state(!this.acknowledge, "'acknowledge' must be false when using a socket expression");
```

Comment:
There is another `acknowledge` word in this class which should be fixed with the typo - `acknowlege`.
In the `handleMessageInternal()`.